src/gui/uithememanager.cpp

```diff
@@ -58,6 +61,88 @@ namespace
 
         return {};
     }
+
+    QByteArray readFile(const QString &fileName)
+    {
+        QFile file {fileName};
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+        {
+            LogMsg(UIThemeManager::tr("UITheme - Failed to open \"%1\". Reason: %2")
+                    .arg(QFileInfo(fileName).fileName(), file.errorString())
+                   , Log::WARNING);
+            return {};
+        }
+
+        return file.readAll();
+    }
+
+    class QRCThemeSource final : public UIThemeSource
+    {
+    public:
+        QByteArray styleSheet() override
+        {
+            return readFile(QRC_THEME_DIR + STYLESHEET_FILE_NAME);
+        }
+
+        QByteArray config() override
+        {
+            return readFile(QRC_THEME_DIR + CONFIG_FILE_NAME);
+        }
+
+        QString iconPath(const QString &iconId) override
+        {
+            return findIcon(iconId, QRC_ICONS_DIR);
```

Comment:
```suggestion
            return findIcon(iconId, (QRC_THEME_DIR + THEME_ICONS_DIR));
```

Reply:
`iconPath` will be called many times while other methods of the class where the string is dynamically created only called once, hence the separate variable for this specific method.